sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.cs

```diff
@@ -74,5 +75,25 @@ private static QuickPulseSDKClientAPIsRestClient InitializeRestClient(LiveMetric
 
             return new QuickPulseSDKClientAPIsRestClient(new ClientDiagnostics(options), pipeline, host: connectionVars.LiveEndpoint);
         }
+
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposedValue)
+            {
+                if (disposing)
+                {
+                    ShutdownState();
+                }
+
+                _disposedValue = true;
+            }
+        }
```

Comment:
Do we expect to ship this class publicly or have more derived implementations? If not FYI you could seal the class and do the more simple form:

```csharp
internal sealed partial class Manager : IDisposable
{
        public void Dispose()
        {
            if (!_disposedValue)
            {
                 ShutdownState();
                _disposedValue = true;
            }
        }
}
```

Why?

That dispose pattern you have is for public classes which may become bases classes to more derived logic. If a user makes a more specific implementation they may need a finalizer to ensure clean up of unmanaged resources and they may need to clean up managed resource during dispose. That's why the `GC.SuppressFinalize` is there and why `protected virtual void Dispose(bool disposing)` is there. If you seal the class, you eliminate those concerns.

Reply:
I always let Visual Studio generate the Dispose pattern, but I definitely prefer the simpler form.
Thanks for the learning!